### PR TITLE
adding reference kernels for arithmetic reduction ops

### PR DESCRIPTION
**Pull request by adstraw**

### CODE REVIEW

File: src/ngraph/op/max.cpp [hunk 1]
```diff
@@ -82,6 +84,55 @@ shared_ptr<Node> op::v0::Max::get_default_value() const
     }
 }
 
+namespace
+{
+    template <element::Type_t ET>
+    bool evaluate(const HostTensorPtr& arg, const HostTensorPtr& out, const AxisSet& axes)
+    {
```
Comment (diyessi) [inline src/ngraph/op/max.cpp:91]: Need to set the output shape/element type. See what `Add` does.

File: src/ngraph/runtime/interpreter/opset_int_tbl.hpp [hunk 2]
```diff
@@ -24,6 +24,8 @@ NGRAPH_OP(LogicalAnd, op::v1)
 NGRAPH_OP(LogicalOr, op::v1)
 NGRAPH_OP(LogicalXor, op::v1)
 NGRAPH_OP(LogicalNot, op::v1)
+NGRAPH_OP(ReduceMax, op::v1)
```
Comment (diyessi) [inline src/ngraph/runtime/interpreter/opset_int_tbl.hpp:27]: Don't need these; evaluate will pick them up automatically.

File: src/ngraph/runtime/interpreter/int_executable.hpp [hunk 3]
```diff
@@ -1593,14 +1573,18 @@ class INTERPRETER_BACKEND_API ngraph::runtime::interpreter::INTExecutable : publ
...
         case OP_TYPEID::Maximum:
+        case OP_TYPEID::Min:
         case OP_TYPEID::Minimum:
         case OP_TYPEID::Multiply:
         case OP_TYPEID::NonZero_v3:
         case OP_TYPEID::NotEqual:
         case OP_TYPEID::Or:
         case OP_TYPEID::Power:
         case OP_TYPEID::Range:
+        case OP_TYPEID::ReduceMax_v1:
```
Comment (diyessi) [inline src/ngraph/runtime/interpreter/int_executable.hpp:1586]: Don't need these when you take them out out opset_in_tbl.

File: src/ngraph/runtime/interpreter/opset_int_tbl.hpp (hunk 2 above)
Comment (adstraw) [reply src/ngraph/runtime/interpreter/opset_int_tbl.hpp]: Done

File: src/ngraph/runtime/interpreter/int_executable.hpp (hunk 3 above)
Comment (adstraw) [reply src/ngraph/runtime/interpreter/int_executable.hpp]: Done

File: src/ngraph/op/max.cpp (hunk 1 above)
Comment (adstraw) [reply src/ngraph/op/max.cpp:92]: Done

**Comment by adstraw** [comment]
TESTNOW

File: src/ngraph/op/max.cpp [hunk 4]
```diff
@@ -97,3 +147,8 @@ shared_ptr<Node> op::v1::ReduceMax::clone_with_new_inputs(const OutputVector& ne
     check_new_args_count(this, new_args);
     return make_shared<op::v1::ReduceMax>(new_args.at(0), new_args.at(1), get_keep_dims());
 }
+
+bool op::v1::ReduceMax::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs)
+{
+    return evaluate_max(inputs[0], outputs[0], get_reduction_axes());
+}
```
Comment (rkimballn1) [inline src/ngraph/op/max.cpp:154]:
```suggestion
}

```

File: test/backend/reduce_max.in.cpp [hunk 5]
```diff
@@ -0,0 +1,855 @@
...
+    // Create some tensors for input/output
+    Shape shape_a{3, 2};
+    auto a = backend->create_tensor(element::f32, shape_a);
+    copy_data(a, vector<float>{1, 2, 3, 4, 5, 6});
+    auto result = backend->create_dynamic_tensor(element::f32, PartialShape::dynamic());
+
+    auto handle = backend->compile(f);
+    handle->call_with_validate({result}, {a});
+    EXPECT_TRUE(test::all_close_f((vector<float>{2, 4, 6}), read_vector<float>(result)));
+}
```
Comment (rkimballn1) [inline test/backend/reduce_max.in.cpp:855]:
```suggestion
}

```

File: src/ngraph/op/min.cpp [hunk 6]
```diff
@@ -97,3 +147,8 @@ shared_ptr<Node> op::v1::ReduceMin::clone_with_new_inputs(const OutputVector& ne
     check_new_args_count(this, new_args);
     return make_shared<op::v1::ReduceMin>(new_args.at(0), new_args.at(1), get_keep_dims());
 }
+
+bool op::v1::ReduceMin::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs)
+{
+    return evaluate_min(inputs[0], outputs[0], get_reduction_axes());
+}
```
Comment (rkimballn1) [inline src/ngraph/op/min.cpp:154]:
```suggestion
}

```

**Comment by adstraw** [comment]
TESTNOW

**Pull request merged**